Release notes 4.6 (#2049)
ibexa/documentation-developer: docs/release_notes/ibexa_dxp_v4.6.md

This release adds new endpoints that allow you to manage orders by using REST API:

- GET `/shipments` -  loads a list of shipments
- GET `/shipments/{identifier}` - loads a single shipment based on its identifier
- PATCH `/shipments/{identifier}` - updates a shipment
- GET `/shipping/methods` - loads shipping methods
- GET `/shipping/methods/{identifier}` - loads shipping methods based on their identifiers
- GET `/shipping/method-types` - loads shipping methods types
- GET `/shipping/method-types/{identifier}` - loads shipping methods type based on their identifiers
- GET `/orders/order/{identifier}/shipments` - loads a list of shipments.

#### REST API for company accounts [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]

This release adds new endpoints that allow you to manage companies in your platform with REST API:

- GET `/sales-representatives` - returns paginated list of available sales representatives

#### PHP API 

### 

### 

### Ibexa Connect

For list of changes in Ibexa Connect, see [Ibexa app release notes](https://doc.ibexa.co/projects/connect/en/latest/general/ibexa_app_release_notes/).

> 
> ### Deprecations

Reviewer: ```suggestion

### DDEV

[Ibexa DXP can officially be run on DDEV](https://ddev.readthedocs.io/en/latest/users/quickstart/#ibexa-dxp). To run on DDEV a licensed edition of Ibexa DXP, see [Install with DDEV](install_with_ddev.md).

### Deprecations
```

Reviewer: ```suggestion

### DDEV

[Ibexa DXP can officially be run on DDEV](https://ddev.readthedocs.io/en/latest/users/quickstart/#ibexa-dxp). For more information, see the [DDEV guide](https://doc.ibexa.co/en/master/getting_started/install_with_ddev/), which offers a step-by-step walkthrough for installing Ibexa DXP.

### Deprecations
```

### New features and improvements in PIM

#### Virtual products

With this feature, you can create virtual products - non-tangible items such as memberships, services, warranties.
To create a virtual product, first, you have to create a virtual Product Type.
Virtual products don’t require shipment when they are purchased without other physical products.

For more information, see [Create virtual products](https://doc.ibexa.co/projects/userguide/en/master/pim/create_virtual_product/.)

#### IsVirtual Criterion

Product search now supports product virutal and physical product type:

- `IsVirtual` - searches for virtual or physical products. 

#### Product page URLs

When you are creating new product type you can set up product URL alias name pattern.
With this feature, you can also create custom URL and URL alias name pattern field based on product attributes.
Customized URLs are easy to remember, help with SEO optimization and reduce bounce rates on the website. 

For more information, see [Product page URLs](https://doc.ibexa.co/projects/userguide/en/master/pim/work_with_product_page_urls/).

> 

Reviewer: [WIP]
```suggestion

#### Product migration

[Product variant](importing_data.md#product-variant) and [product asset](importing_data.md#product-asset) can be created through [data migration](data_migration.md).

```

Reviewer: Nothing to do w/ perso, misplaced since reordering.

@dabrt Where could I move this entry?

Reviewer: Done in 992a8d6095dd1e9982a202c395cc8674d7999fef

---
description: Ibexa DXP v4.6 brings improvements to Commerce, PIM and Personalization offerings, and a number of changes in CDP and Ibexa Connect.
---

<!-- vale VariablesVersion = NO -->

# Ibexa DXP v4.6

**Version number**: v4.6

> **Release date**: July 20, 2023

Reviewer 2: As mentioned on Slack, this must be fixed.

Reviewer 3: Once we know the date, we will change it

**Release type**: [LTS](https://support.ibexa.co/Public/service-life)

**Update**: [v4.5.x to v4.6](https://doc.ibexa.co/en/latest/update_and_migration/from_4.5/update_from_4.5/)

## Notable changes

### Ibexa Headless

[[= product_name_content =]] changes name to [[= product_name_headless =]] to emphasize [[= product_name_base =]]'s capacity for headless architecture.

The feature set and capabilities of the product remain the same.

### UX and UI improvements

Several improvements to the Back Office interface enhance the user experience.

#### Customizable dashboard

Users can now customize the dashboard depending on their needs.

> This solution uses an online editor - Dashboard Builder.

Reviewer 4: Not needed in RN. Explained in the article.

It improves productivity, allows to enhance the default dashboard with additional widgets,
and helps to make better business decisions based on data.

![Customizable dashboard](img/4.6_customizable_dashboard.png)

For more information, see [Customizable dashboard](https://doc.ibexa.co/projects/userguide/en/master/getting_started/discover_ui/#customizable-dashboard).

> 

Reviewer 4: Link corrected.

- **Undo** and **Redo** buttons.

![Page Builder interface](img/4.6_page_builder_interface.png)

For more information, see [Page Builder interface](https://doc.ibexa.co/projects/userguide/en/master/content_management/create_edit_pages/#page-builder-interface).

#### Editing embedded Content items

User can now edit embedded Content items without leaving current window.
This function is available in the Rich Text Field when creating Content items, for selected blocks in the Page Builder,
and while adding or modifying a Content relation.

![Editing embedded Content items](img/4.6_editing_embedded_content_items.png)

For more information, see [Edit embedded Content items](https://doc.ibexa.co/projects/userguide/en/master/content_management/create_edit_content_items/#edit-embedded-content-items).

#### Focus mode

With multiple changes to the Back Office UI intended to expose the most important information and actions, editors can now better focus on their work.
For more information, see [Focus mode](https://doc.ibexa.co/projects/userguide/en/latest/getting_started/discover_ui/#focus-mode).

![Focus mode](img/4.6_focus_mode.png "Focus mode")

> As part of this effort, some other changes were introduced that apply to both regular and Focus mode:

Reviewer 5: Since we mention the twig function added for testing remote pim, we shoul mention here the one added to test if focus mode is active or not.

- In Content item details view, tabs have been reordered by their relevance
- **Authors** and **Sub-items** are now separate tabs in Content item details view
- Former **Details** tab is now called **Technical details** and changed its position
- Preview is available in many new places, such as the **View** tab in Content item details view, or as miniatures when you hover over the Content Tree

> 

Reviewer 3: I am not listing Focus mode features here, they are available when you follow the link above (line 73).
These bullets mention changes outside the scope of this epic but made as part of it.

![Distraction free mode](img/4.6_distraction_free_mode.png)

For more information, see [Distraction free mode](https://doc.ibexa.co/projects/userguide/en/master/content_management/create_edit_content_items/#distraction-free-mode).

#### Simplified user actions

Button text now precisely describes actions, so that users who create or edit content understand the purpose of each button.

![Improved button text](img/4.6_publishing_options.png)

#### Draft section added to Content

For streamlining purpose, the **Draft** section is now situated under **Content**.
Users can now easily find and manage their drafts and published content from one area.

![Draft section added to Content](img/4.6_drafts.png)

#### User profile and new options in user settings

With personal touch in mind, editors can now upload their photos, and provide the following information in their user profiles:

- Email
- Department
- Position
- Location
- Signature

Also, editors and other users can customize their experience even better, with new configuration options that have been added to user settings.

For more information, see [user profile and settings documentation]().

> #### Recent activity log

Reviewer 5: I feel like this feature deserves a screenshot :)
We could also mention that new policies can be used to determine if admin users can see their own activities or everybody's. But maybe it's too specific for the release notes.

and how to [customize autocompletion suggestions](https://doc.ibexa.co/en/master/docs/administration/back_office/search_suggestion/).

##### Filtering and sorting

The search result page can be sorted in other orders than relevance. Name, publication of modification dates, this can be extended.

Filters can be applied to the search page to narrow down the results.

For more information,
see [user documentation](https://doc.ibexa.co/projects/userguide/en/master/search/search_for_content/#filtered-search),
and how to [customize search sorting](https://doc.ibexa.co/en/master/docs/administration/back_office/search_sorting/.

#### New and updated Content Type icons

To help users quickly identify different content types in the Back Office, all Content Type references are now accompanied with icons.
Also, Content Type icons have changed slightly.

![Content type icons](img/4.6_content_type_icons.png)

### Ibexa Image picker

Editors can now use a Digital Asset Management platform that enables storing media assets in a central location, as well as organizing, distributing, and sharing them across many channels.

For more information, see [link to docs]().

> ### New features and improvements in PIM

Reviewer 3: Headless (former Content) is the lowest edition of the product, which means that all editions have PIM. No need for the badge.

> ### New features and improvements in PIM

Reviewer 6: I have an impression, that some people still don't know that PIM is offered from the lowest edition.

#### Remote PIM support

> Historically, [[= product_name =]] was configured to fetch objects and products from a local database, where they were created through the Back Office or with the REST API.

Reviewer 5: I feel like this part is a bit too detailed for the release notes.

- use Customer Groups to apply different prices to products,
- define and use currencies.

For more information about Remote PIM support and the solution's limitations, see [PIM product guide](pim_guide.md#limitations).

#### Virtual products

With this feature, you can create virtual products - non-tangible items such as memberships, services, warranties.
To create a virtual product, first, you have to create a virtual product type.
Virtual products don’t require shipment when they're purchased without other physical products.

For more information, see [Create virtual products](https://doc.ibexa.co/projects/userguide/en/master/pim/create_virtual_product/).

#### Product page URLs

When you are creating a new product type, you can set up a product URL alias name pattern.
With this feature, you can also create custom URL and URL alias name pattern field based on product attributes.
Customized URLs are easier to remember, help with SEO optimization and reduce bounce rates on the website.

For more information, see [Product page URLs](https://doc.ibexa.co/projects/userguide/en/master/pim/work_with_product_page_urls/).

#### VAT assignment moved to a new place

> Users who are creating or editing a product type are less likely to forget about setting VAT rates, because they now have a more prominent place.

Reviewer 6: @dabrt can we also change this "prominent place"? :)
And the screenshot will be replaced by "4.6_catalog_vat_rates.png", right?

For more information, see [Create product types](https://doc.ibexa.co/projects/userguide/en/master/pim/create_product_types/).

#### Updated VAT configuration

VAT rates configuration has been extended to accept additional flags under the `extras` key.
Developers can use them, for example, to pass additional information to the UI, or define special exclusion rules.

For more information, see [VAT rates](https://doc.ibexa.co/en/master/pim/pim_configuration/#vat-rates).

##### New Twig functions

The `ibexa_is_pim_local` Twig helper has been introduced, which can be used in templates to [check whether product data comes from a local or remote data source](storefront_twig_functions.md#ibexa_is_pim_local), and adjust their behavior accordingly.
Also, several new Twig functions have been implemented that help [get product availability information](product_twig_functions.md#ibexa_has_product_availability).

##### New and modified query types

The `ProductContentAwareListQueryType` has been created to allow finding products that come from a local database, while `ProductListQueryType` has been modified to find products from an external source of truth.

> ##### New Search Criterion

Reviewer 4: Capitalization used to be consistent with the entire section: https://doc.ibexa.co/en/latest/search/criteria_reference/search_criteria_reference/

With `IsVirtual` criterion that searches for virtual or physical products, product search now supports products of virtual and physical type.

#### Product migration

[Product variants](importing_data.md#product-variants) and [product assets](importing_data.md#product-assets) can now be created through [data migration](data_migration.md).

###  New features and improvements in Commerce [[% include 'snippets/commerce_badge.md' %]]

> 

Reviewer 6: We forgot about one very important feature for Commerce: **Integration with payment gateways** https://issues.ibexa.co/browse/IBX-5713

#### Customize checkout workflow

You can create a PHP definition of the new strategy that allows for workflow manipulation.
Defining strategy allows to add conditional steps for workflow if needed.
When a conditional step is added, the checkout process uses the specified workflow and proceeds to the defined step.

For more information, see [Create custom strategy](https://doc.ibexa.co/en/master/commerce/checkout/customize_checkout/#create-custom-strategy).

#### Manage multiple checkout workflows

When working with multiple checkout workflows, you can now specify the desired workflow by passing its name as an argument to the checkout initiation button or link.

For more information, see [Manage multiple workflows](https://doc.ibexa.co/en/master/commerce/checkout/customize_checkout/#manage-multiple-workflows).

#### Adding context data to cart

Attach context data to both the Cart and its individual Cart Entries.
This feature enhances the flexibility and customization of your e-commerce application,
enabling you to associate additional information with your cart and its contents.
By leveraging context data, such as promo codes or custom texts,

> you can tailor the e-commerce experience for your customers and enhance the capabilities of your application.

Reviewer 6: ```suggestion
you can tailor the shopping experience for your customers and enhance the capabilities of your application.
```

For more information, see [Email triggers](https://doc.ibexa.co/projects/userguide/en/4.5/personalization/triggers.md).

#### Multiple attributes in recommendation computation

With this feature, you get an option to combine several attribute types when computing recommendations.
As a result, users can be presented with recommendations from an intersection of submodel results.

For more information, see [Submodel parameters](recommendation_api.md#submodel-parameters) and [Submodels](https://doc.ibexa.co/projects/userguide/en/latest/personalization/recommendation_models/#submodels).

#### New scenario filter

Depending on a setting that you make when defining a scenario, the recommendation response can now include either product variants or base products only.
This way you can deliver more accurate recommendations and avoid showing multiple variants of the same product to the client.

For more information, see [Commerce-specific filters](https://doc.ibexa.co/projects/userguide/en/latest/personalization/filters/#commerce-specific-filters).

## Other changes

### Expression Language

> New `project_dir()` expression language function that allows you to reference current project directory.

Reviewer 7: I think it's worth mentioning that it's for Migrations

### Site Factory events

> Site Factory events have been moved from the `lib` directory to `contracts`.

Reviewer 7: I think it would be better to mention namespaces, not directories - they were moved to `Ibexa\Contracts\SiteFactory\Events`. We might mention that it's done in a BC-safe manner.

POV ping @mikadamczyk

Reviewer 8: It could be:
Site Factory events have been moved from the `Ibexa\SiteFactory\ServiceEvent\Events` namespace to the `Ibexa\Contracts\SiteFactory\Events` namespace, keeping a BC-safe manner.

For a full list of events, see [Site events](https://doc.ibexa.co/en/latest/api/event_reference/site_events/).

Event handling system was improved with the addition of listeners based on `CreateSiteEvent`, `DeleteSiteEvent`, and `UpdateSiteEvent`.
New listeners automatically grant permissions to log in to a site, providing a more seamless site management experience.

### Integration with Actito

By using the Actito gateway you can send emails to the end-users about changes in the status of various operations in your commerce presence.

### Integration with Qualifio Engage

> Use Qualifio Engagne integration to create engaging marketing experiences to your customers.

Reviewer 7: ```suggestion
Use Qualifio Engage integration to create engaging marketing experiences to your customers.
```
It's a typo I think?

### Integration with SeenThis!

> Integration with SeenThis! service provides an adaptive streaming technology with no limitations as conventional streaming service.

Reviewer 2: ```suggestion
Integration with SeenThis! service provides an adaptive streaming technology with no limitations, unlike conventional streaming services.
```
Unclear sentence. Was this what was meant?

> It allows to preserve the best video quality with minimum amount of data transfer.

Reviewer 2: ```suggestion
It allows you to preserve the best video quality with the minimum amount of data transfer.
```

For more information, see [SeenThis! block](https://doc.ibexa.co/projects/userguide/en/latest/content_management/block_reference/#sales-representative).

### API improvements

#### REST API

> ##### New method signature

Reviewer 5: I'd move this one to the bottom of the REST API chapter as it is really more low level.

The method signature for `\Ibexa\Contracts\Rest\Output\Generator::startValueElement` has been updated to the following:

```php
    /**
     * @phpstan-param scalar $value
     * @phpstan-param array<string, scalar> $attributes
     */
    abstract public function startValueElement(string $name, $value, array $attributes = []): void;
```

Any third party code that extends `\Ibexa\Contracts\Rest\Output\Generator` needs to update the method signature accordingly.

##### REST API for shipping [[% include 'snippets/commerce_badge.md' %]]

> Endpoints that allow you to manage orders by using REST API:

Reviewer 6: ```suggestion
Endpoints that allow you to manage shipping methods and shipments by using REST API:
```

For more information, see [Email triggers](https://doc.ibexa.co/projects/userguide/en/4.5/personalization/triggers.md).

#### Multiple attributes in recommendation computation

With this feature, you get an option to combine several attribute types when computing recommendations.
As a result, users can be presented with recommendations from an intersection of submodel results.

For more information, see [Submodel parameters](recommendation_api.md#submodel-parameters) and [Submodels](https://doc.ibexa.co/projects/userguide/en/latest/personalization/recommendation_models/#submodels).

#### New scenario filter

Depending on a setting that you make when defining a scenario, the recommendation response can now include either product variants or base products only.
This way you can deliver more accurate recommendations and avoid showing multiple variants of the same product to the client.

For more information, see [Commerce-specific filters](https://doc.ibexa.co/projects/userguide/en/latest/personalization/filters/#commerce-specific-filters).

## Other changes

### Expression Language

> New `ibexa.migrations.template.project_dir()` expression language function that allows you to reference current project directory.

Reviewer 7: ```suggestion
New `project_dir()` expression language function that allows you to reference current project directory while using Ibexa Migrations
```

The function name was good, it's just that the sentence was missing context where this new function can be used - it's not for PHP code, it's only when writing the YAML Migration files.

#### Focus mode

With multiple changes to the Back Office UI intended to expose the most important information and actions, editors can now better focus on their work.
The UI is now more friendly and appealing for marketers and editors, with simplified Content structure, designed with new and non-advanced users in mind.
For more information, see [Focus mode](https://doc.ibexa.co/projects/userguide/en/latest/getting_started/discover_ui/#focus-mode).

> 

Reviewer 3: https://doc.ibexa.co/projects/userguide/en/master/getting_started/discover_ui/#focus-mode
